### ISSUE TITLE
tl failed to resolve map value type with some format key

### ISSUE THREAD

**Issue by virusdefender** [teal-language/tl#332]
source code is

```
local keystr = "aaaa"

local record user_login_count_t
end

local data:{string: user_login_count_t} = {
    key={},
    [keystr]={}
}

for k, v in pairs(data) do
    print(k, v)
end
```

the error message is

```
local declaration: data: got {}, expected user_login_count_t
```

**Comment by lenscas** [comment]
either remove the `[]` around keystr or quote it (`["keystr"]`). What you currently have wouldn't be valid lua.
```
lua
Lua 5.4.2  Copyright (C) 1994-2020 Lua.org, PUC-Rio
> {[keystr]={}}
stdin:1: table index is nil
stack traceback:
	stdin:1: in main chunk
	[C]: in ?
> 
```

**Comment by euclidianAce** [comment]
@lenscas That is valid lua, using the `[]` syntax lets you define a key with an arbitrary expression, and if that expression is `nil`, you get a runtime error like you did
```
> keystr = "a"; print({[keystr] = {}})
table: 0x55b820e68f20
```
```
> keystr = nil; print({[keystr] = {}})
stdin:1: table index is nil
stack traceback:
        stdin:1: in main chunk
        [C]: in ?
```

**Comment by hishamhm** [comment]
@virusdefender thanks for reporting! fix in #333, please let me know if it fixes it for you!
